ducktape/tests/loader.py

```diff
@@ -192,16 +229,24 @@ def import_modules(self, file_list):
 
         return module_list
 
-    def expand_module(self, module):
+    def expand_module(self, module_and_file):
```

Comment:
This and the dict are kind of awkward. Why not choose either a concrete named type (with expected structure) or separate this into two separate parameters?

It might also help to rename some of these methods to have a `_` prefix so we know when they are internal (and therefore it might not be *as* bad for them to have somewhat weird arguments like this one).